CONTRIBUTION.md

```diff
@@ -4,78 +4,77 @@ When contributing to this repository, please first discuss the change you wish t
-3. Make sure this repository is set as the [upstream remote repository](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork) so you are able to fetch the latest commits.
-4. Push all your changes to the `develop` branch of the forked repository.
+### The development environment setup
 
-*Note*: Make sure you have you have the latest version of the `develop` branch on your local machine.
+For setting up your environment, please follow the instructions in the `README.md` under `Installation Guide For: Contributors`.
 
-```
-git checkout develop
-git pull upstream develop
-```
+### The development workflow
 
-5. Create pull requests to the upstream repository.
+For new features, bugs, enhancements:
 
-### The development lifecycle
+#### 1. Branch Setup
+* Pull the latest code from the develop branch in the upstream repository.
+* Checkout a new branch formatted like so: `develop-<feature/fix-name>` from the develop branch
 
-1. Pull the latest content from the `develop` branch of this central repository (not your fork).
-2. Create a branch off the `develop` branch. Name the branch appropriately, either briefly summarizing the bug (ex., `spatil/add-restapi-layer`) or feature or simply use the issue number in the name (ex., `spatil/issue-414-fix`).
-3. After completing work and testing locally, push the code to the appropriate branch on your fork.
-4. In Github, create a pull request from the bug/feature branch of your fork to the `develop` branch of the central repository.
+#### 2. Development Workflow
+* Develop on your new branch.
+* Ensure pyproject.toml and poetry.lock files are compatible with your environment.
+* Add changed files for tracking and commit changes using [best practices](https://www.perforce.com/blog/vcs/git-best-practices-git-commit)
+* Have granular commits: not “too many” file changes, and not hundreds of code lines of changes
+* You can choose to create a draft PR if you prefer to develop this way
 
-> A Sage Bionetworks engineer must review and accept your pull request. A code review (which happens with both the contributor and the reviewer present) is required for contributing.
+#### 3. Branch Management
+* Push code to `develop-<feature/fix-name>` in upstream repo:
+  ```
+  git push <upstream> develop-<feature/fix-name>
+  ```
+* Branch off `develop-<feature/fix-name>` if you need to work on multiple features associated with the same code base
+* After feature work is complete and before creating a PR to the develop branch in upstream
+    a. ensure that code runs locally
+    b. test for logical correctness locally
+    c. wait for git workflow to complete (e.g. tests are run) on github
 
-### Development environment setup
+#### 4. Pull Request and Review
```

Comment:
Should we include a reminder to run pre-commit manually on all the files if the hooks weren't enabled?